src/fmap.c

```diff
@@ -0,0 +1,70 @@
+// Flashmap support.
+//
+// Copyright (C) 2019 3mdeb
+//
+// This file may be distributed under the terms of the GNU LGPLv3 license.
+
+#include "malloc.h" // malloc
+#include "string.h" // memcpy
+#include "output.h" // dprintf
+#include "fmap.h"   // find_fmap_directory
+
+static void* fmap_entry = NULL;
+
+#define ROM_BEGIN       ((void *)0xFF800000)
```

Comment:
We should use lib_sysinfo.spi_flash.size to determine the ROM_BEGIN value. This is dirty hack as we know we have a 8MB SPI flash. The ROM_BEGIN should be calculated in code with following formula:

ROM_BEGIN = 0xFFFFFFFF - lib_sysinfo.spi_flash.size + 1

CHeck whether it is possible to access such data from lib_sysinfo.